### ISSUE TITLE
MDX formatter adding semicolon to JSX code blocks

### ISSUE THREAD

**Issue by danoc** [prettier/prettier#5374]
**Prettier 1.15.1**
(Seems like the playground doesn't support MDX yet?)
```sh
# Options (if any):
Default options
```

**Input:**
`test.mdx`:
````mdx
This is a MDX file!

```jsx
<Button />
``` 
````
**Output:**
````mdx
This is a MDX file!

```jsx
<Button />;
```
````

**Expected behavior:**
````mdx
This is a MDX file!

```jsx
<Button />
```
````

This seems related to https://github.com/prettier/prettier/issues/3196 and https://github.com/prettier/prettier/pull/3330.

***
Thanks for the help!

**Comment by duailibe** [comment]
> Seems like the playground doesn't support MDX yet?

Sure it works! Try a hard refresh to clear the cache

**Prettier 1.15.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAVAFgSwM4AIe4CGuAsgCIAauAZpgDZwCEAOlKwAacBW2AHqwB4AQgFcYMaLgD0APg6dcrEABoQEAA4xM0bMlCEATgYgB3AAqGEulIQBuETABMVIAEYHCYANZwYAZXVPTCgAc2QYAxE4VXQYAFs6AHUseGxAsDg-K0wtWxyAT2RwbF1VYOw4AxgzDxC4wmRqQjoK1R5eIQ9vXz9COLgAGWC4RubWkHa-YJCGAEURCHhRluiQQIMKgyK4x14XdQNgmESnGHRkAA4ABlUDiArEj3Uig7hN2xHVAzgARxFMb41Qh1BpIJorVQVOKYcKRVbYaZzBZLMFjVYwQiuE6OM7IABMqgihHo0wAwhA4vUilBoJ8QCIKqhMdZwRUAL5soA)
```sh
--parser mdx
```

**Input:**
````mdx
This is a MDX file!

```jsx
<Button />
``` 

````

**Output:**
````mdx
This is a MDX file!

```jsx
<Button />;
```

````

**Comment by j-f1** [comment]
That’s not quite the MDX syntax:

**Prettier 1.15.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAVAFgSwM4AIe4CGuAsgCIAauAZpgDZwCEAOlKwDwBCArjDNLgD0APlYgANCAgAHGJmjZkoQgCcVEAO4AFVQkUpCANwiYAJhJAAjFYTABrODADK025igBzZDBXc4k9BgAWzoAdSx4bFcwOCc9TDlDBIBPZHBsRUl3bDgVGC0bDyDCZGpCOhzJACtsAA9OG3tHJ0IguAAZdzhS8sqQGtqndw8GAEVuCHgeiv8QVxUclTSg01qLaRV3GFCzGHRkAA4ABkkNiBzQm2k0jbhFw27JFTgAR25MZ4LCIpKkMpnJDkgphvL5ZthhmMJlM-r1ZjBCJYdqY9sgAEySHyEejDADCECCxTSUGgjxA3ByqER+n+OQAvnSgA)
```sh
--parser mdx
```

**Input:**
```mdx
This is a MDX file!

<Button />

```

**Output:**
```mdx
This is a MDX file!

<Button />

```
(Your code inserts a JSX-formatted code block instead of rendering the JSX)

**Comment by danoc** [comment]
Ah, yes. Sorry if that wasn't originally clear.

**Comment by j-f1** [comment]
For clarity, with regular Markdown:

**Prettier 1.15.1**
[Playground link](https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAVAFgSwM4AIe4CGuAVCQLKEBOA1gCYQDuUZuAZpgDZwCEAOlAEADEQCtsADwEAeAEIBXGDGi4A9AD5hI3AJAAaEBAAOMTNGzJQ1KkwAK1BBZSEAbhEx19IAEZVCYGjgYAGUjf0woAHNkGCp5OAN0GABbTgB1LHhsMLA4YMdMUxdCgE9kcGwLAwjsOCoYWz9I5MJkNkJOWoNxCVk-AKDgwmS4ABkIuDaOrpAe4IjI7gBFeQh4Kc6EkDCqWqpyltoGZi8jKgiYNI8YdGQADgAGAzOIWrS-I3KzuD2XSYMqHAAI7yTCAxqEZqtJDtTYGWrJTAxOJbbALZardYw6ZbGCEbxXOg3ZAAJgMsUIXAWAGEIMkWuUoNB-iB5LVUPinLDagBfHlAA)
```sh
--parser markdown
```

**Input:**
````markdown
This is a **Markdown** file!

```jsx
<Button />
``` 

````

**Output:**
````markdown
This is a **Markdown** file!

```jsx
<Button />
```

````

**Comment by ikatyang** [comment]
The fix is available in [`1.15.2`](https://github.com/prettier/prettier/blob/master/CHANGELOG.md#1152).
